files/solutions/tictactoe_easy.cpp

```diff
@@ -0,0 +1,133 @@
+#include <iostream>
+#include <array>
+
+std::string fts(int feld) {
+
+    if (feld == 0) {
+        return " ";
+    } else if (feld == 1) {
+        return "X";
+    } else if (feld == 2) {
+        return "O";
+    } else {
+        return "?";
+    }
+}
+
+int frage_feld_nummer(std::array<int, 9> feld){
+    int in = -1;
+    bool valid = false;
+    while (!valid) {
+        std::cout << "Gib ein Feld ein (0-8): ";
+        std::cin >> in;
+        if (in < 0 || in > 8) {
+            std::cout << "Ungültige Feldnummer!" << std::endl;
+        }
+        if (feld[in] != 0) {
+            std::cout << "Feld ist besetzt!" << std::endl;
+        } 
+        else {
+            valid = true; // Markiert die Eingabe als gültig wenn keine Fehler aufgetauch sind
+        }
+    }
+    return in;
+}
+
+void gebe_feld_aus(std::array<int, 9> feld){
+    std::cout << "   |   |   " << std::endl;
+    std::cout << " " << fts(feld[0]) << " | " << fts(feld[1]) << " | " << fts(feld[2]) << " " << std::endl;
+    std::cout << "   |   |   " << std::endl;
+    std::cout << "---+---+---" << std::endl;
+    std::cout << "   |   |   " << std::endl;
+    std::cout << " " << fts(feld[3]) << " | " << fts(feld[4]) << " | " << fts(feld[5]) << " " << std::endl;
+    std::cout << "   |   |   " << std::endl;
+    std::cout << "---+---+---" << std::endl;
+    std::cout << "   |   |   " << std::endl;
+    std::cout << " " << fts(feld[6]) << " | " << fts(feld[7]) << " | " << fts(feld[8]) << " " << std::endl;
+    std::cout << "   |   |   " << std::endl;
+}
+
+int gewinnerin(std::array<int, 9> feld){
+   // Prüfe Reihen und Spalten
+    for (int i = 0; i < 3; i++) {
+        int a = feld[3*i] & feld[3*i+1] & feld[3*i+2];
+        int b = feld[3*i] | feld[3*i+1] | feld[3*i+2];
+        if (a == b && a != 0) {
+            return feld[3*i];
+        }
+        a = feld[i] & feld[i+3] & feld[i+6];
+        b = feld[i] | feld[i+3] | feld[i+6];
+        if (a == b && a != 0) {
+            return feld[i];
+        }
+    }
+    // Prüfe Diagonale 1
+    int a = feld[0] & feld[4] & feld[8];
+    int b = feld[0] | feld[4] | feld[8];
+    if (a == b && a != 0) {
+        return feld[0];
+    }
+
+    // Prüfe Diagonale 2
+    a = feld[2] & feld[4] & feld[6];
+    b = feld[2] | feld[4] | feld[6];
+    if (a == b && a != 0) {
+        return feld[2];
+    }
+
+    for (int i = 0; i < 9; i++) {
+        if (feld[i] == 0) {
+            return 0;
+        }
+    }
+    return 3;
+}
+
+
+
+int main() {
+    // Setup
+    std::array<int, 9> spielfeld{0};
```

Comment:
This does currently not compile on Apple LLVM/Clang as they have not yet fully implemented the C++ 20 Standard. As normal LLVM-Clang has implemented aggregate Initialisation as far back as 2022 and g++ supports this as well, I have decided to ignore this issue. 
A possible fix could be to just initialise it using a for-loop, but I’m lazy so instead of fixing this by writing 3 lines I will write 20 lines while ranting here

Reply:
I'd open an Issue for this and wait how far the homebrew clang got next year